manifests/charts/istio-control/istio-discovery/templates/rolebinding.yaml

```diff
@@ -1,16 +1,20 @@
 apiVersion: rbac.authorization.k8s.io/v1
 kind: RoleBinding
 metadata:
-  name: istiod{{- if not (eq .Values.revision "")}}-{{ .Values.revision }}{{- end }}
+  name: istiod{{- if not (eq .Values.revision "")}}-{{ .Values.revision }}{{- end }}{{- if .Values.global.externalIstiod }}-{{ .Values.global.istioNamespace }}{{- end }}
```

Comment:
@Monkeyanator I'm really confused about all the different naming patterns used in these Role/ClusterRole and corresponding binding resources. The ones in base use a pattern`<name>-<namespace>`. The ones in the istiod-remote charts use `<name>[-<rev>]-<namespace>`. The discovery ones are the most confusing. Some use `<name>[-<rev>]`, some use `<name>-cr[-<rev>]`, and some use `<name>-cr[-rev]<namespace>` (note missing `-` before namespace).

Can you help me understand what the naming objectives and restrictions are? Do you think we can change them to one consistent form? Since the external control plane feature is currently Alpha, we can change the istiod-remote ones to whatever makes sense, but I'm wondering how fixed the other ones are. For example is the `revision` feature Beta, and if so, are we stuck with the missing `-` in the `<name>-cr[-rev]<namespace>` names?